### PR TITLE
Fix nested hierarchy in yaml

### LINKED ISSUE #26: bad check/fix multidimensional array
**Which version(s) is affected**: all

**Description**
If you create multidimensional array without array key then checker/fixer don't work correctly

**How to reproduce**
e.g. this yaml
```yaml
patchesJson6902:
-   target:
        group: extensions
        version: v1beta1
        kind: Ingress
        name: shopsys
    path: ./ingress-patch.yaml
```
checker/fixer want fix to this
```yaml
patchesJson6902:
-   target:
        group: extensions
        version: v1beta1
        kind: Ingress
        name: shopsys
        path: ./ingress-patch.yaml
```
and it's incorrect.
See https://github.com/shopsys/shopsys/pull/539#issuecomment-515050285 for better detail

### PR DESCRIPTION
**Pull request by PetrHeinz**
| Q             | A
| ------------- | ---
| Bug fix?                      | yes
| New feature?                  | no
| BC breaks?                    | no
| Standards and tests pass?     | yes
| Fixes issues                  | fixes #26 
| License                       | MIT

Fixing a bug with nested hierarchy, explained in https://github.com/shopsys/shopsys/pull/539#issuecomment-515050285

### CODE REVIEW

File: src/Model/Component/YamlService.php [hunk 1]
```diff
@@ -98,4 +99,44 @@ public static function isLineStartOfArrayWithKeyAndValue($trimmedLine)
...
+
+    /**
+     * line possibly opening an array, e.g. 'foo:' or '- foo:'
+     *
+     * @param string $trimmedLine
+     * @return bool
+     */
+    public static function isLineOpeningAnArray($trimmedLine)
+    {
+        return (bool)preg_match('~^(- +)*(' . Inline::REGEX_QUOTED_STRING . '|[^ \'"{\[].*?) *:$~u', $trimmedLine);
```
Comment (sspooky13) [inline src/Model/Component/YamlService.php:122]: these regexes are killers 😄 but I will hope they work as is described 👍

File: src/Model/YamlIndent/YamlIndentDataFactory.php [hunk 2]
```diff
@@ -217,36 +204,23 @@ private function getCorrectLineForArrayWithKeyAndValue($line, array $fileLines,
...
 
-        /**
-         * solution for more values in array
-         * "- foo: bar"
-         * "  baz: qux:
-         */
-        if (array_key_exists($key + 1, $fileLines) && $this->isNextLineKeyAndValueOfArray($lineWithReplacedDashToSpace, $fileLines[$key + 1])) {
-            $correctIndentsBetweenDashAndKey = $this->getCorrectIndents($countOfIndents - 1); // 1 space is dash, dash is as indent
+        // solution "- foo" (single value of an array)
+        if (!YamlService::isKeyInStartOfString($trimmedLineWithoutDash)) {
```
Comment (sspooky13) [inline src/Model/YamlIndent/YamlIndentDataFactory.php]: hmm you reverse condition, I like it 👍

File: src/Model/Component/YamlService.php (hunk 1 above)
Comment (PetrHeinz) [reply src/Model/Component/YamlService.php:122]: I've pretty much used key matching that is used in Symfony `Yaml\Parser`:

```php
preg_match('#^(?P<key>'.Inline::REGEX_QUOTED_STRING.'|[^ \'"\{\[].*?) *\: ...
```

[see `Symfony\Component\Yaml\Parser` in `v4.3.2`](https://github.com/symfony/symfony/blob/v4.3.2/src/Symfony/Component/Yaml/Parser.php#L191)

Comment (sspooky13) [reply src/Model/Component/YamlService.php:122]: it's ok, just I write my thinks 😄

**Comment by sspooky13** [comment]
Hmm, it's looking good. It's simple and clear.
I hope I can push to your branch. I want to change 1 thing and update changelog, etc.

PS: thanks for a simplify some functionality. More these PR, please 😄

**Comment by PetrHeinz** [comment]
Glad I could've helped. Thanks for the library! :+1:

**Pull request merged**